### PR TITLE
Fix primary key in materialized view

### LINKED ISSUE #62045: "system.tables" doesn't show primary key for materialized views
**Describe the unexpected behavior**
Materialized views don't have primary key in "system.tables"

**How to reproduce**
```sql
CREATE TABLE sample (uid Int16, name String)
ENGINE = MergeTree
PRIMARY KEY uid;

INSERT INTO sample VALUES (1, 'Alice'), (2, 'Bob');

CREATE MATERIALIZED VIEW sample_mv
ENGINE = MergeTree
PRIMARY KEY name
POPULATE AS
SELECT name, uid
FROM sample;

SELECT name, primary_key
FROM system.tables
WHERE name LIKE 'sample%';
-- sample    | uid
-- sample_mv | 
-- [*] for some reason, sample_mv has empty primary_key

SHOW CREATE TABLE sample_mv;
-- CREATE MATERIALIZED VIEW default.sample_mv (
--     `name` String,
--     `uid` Int16
-- )
-- ENGINE = MergeTree
-- PRIMARY KEY name             <- primary key really exists
-- ORDER BY name ...
```
Example: https://fiddle.clickhouse.com/87587364-10d4-4338-b1cc-c6c01f83684e

**Expected behavior**
"system.tables" must show primary key for materialized views in same way as it shows for tables

### PR DESCRIPTION
**Pull request by mxwell**
Fixes #62045

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix displaying of materialized_view primary_key in system.tables. Previously it was shown empty even when a CREATE query included PRIMARY KEY.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
#- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit f43961795a620ffab2e4943fe2f125d14508cdf7 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62319/f43961795a620ffab2e4943fe2f125d14508cdf7/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>❌ failure</td></tr>
<tbody>
</table>

**Comment by jsc0218** [comment]
the clang tidy is not relevant to your pr, probably do a rebase or merge the latest branch would work.

File: src/Storages/StorageMaterializedView.cpp [hunk 1]
```diff
@@ -91,6 +91,12 @@ StorageMaterializedView::StorageMaterializedView(
 {
     StorageInMemoryMetadata storage_metadata;
     storage_metadata.setColumns(columns_);
+    auto * storage_def = query.storage;
```
Comment (jsc0218) [inline src/Storages/StorageMaterializedView.cpp:94]: probably just use `auto` instead of `auto *` to simplify the code here

Comment (mxwell) [reply src/Storages/StorageMaterializedView.cpp:94]: Interestingly, I had `auto storage_def` initially, but some style check complained and suggested to change it to `auto *` to make it more explicit.

Comment (jsc0218) [reply src/Storages/StorageMaterializedView.cpp:94]: Ah, I see, then just use what the style check suggests.

**Review by jsc0218 — APPROVED** [review]

**Pull request merged**